### PR TITLE
Add support for Woodpecker CI 1.x and 2.x

### PR DESCRIPTION
**Pull request by uncomfyhalomacro**
# Breaking Change

This PR supersedes #2335. The changes introduced are the following

## Removed environmental variables required in the struct `Woodpecker`

- `FORGE_URL`. This is not needed based on my usage.
- `woodpecker_repo_link` which uses `CI_REPO_LINK`. The environmental variable `CI_REPO_LINK` is still used if the version is 0.15.0. We just manually change it get `woodpecker_forge_url` to allow compatibility between woodpecker versions.

Reason for removing `woodpecker_repo_link` is that `CI_REPO` already contains `<owner>/<name>`.

## Changed or renamed environmental variables

- `CI_BUILD_EVENT` -> `CI_PIPELINE_EVENT`. 0.15.x used to use the `CI_BUILD_EVENT`. Latest release will use `CI_PIPELINE_EVENT`. Woodpecker 2.0.0 will remove `CI_BUILD*` environmental variables entirely and replace it with `CI_PIPELINE*`. So we are preparing for that transition.

## New environmental variables to be used

- `CI_FORGE_URL`. This is the URL for the constructed forge e.g. `https://github.com` or `https://gitlab.com`. To allow compatibility prior 1.0.0, `CI_REPO_LINK` is used with `match`.
- `CI_SYSTEM_VERSION`. Returns the version number of currently used Woodpecker CI system. This is the environmental variable to be used to check versions.

### CODE REVIEW

**Comment by uncomfyhalomacro** [comment]
Ready for review. Not sure pinging is the correct way to ask for code review. Sorry!

CC: @mortenpi @fredrikekre

**Review by mortenpi — APPROVED** [review]
Apologies for the delay here. I haven't really tested this, but the changes LGTM, and the test suite looks really comprehensive. Thank you!

It's ready to be merged, right? Would you mind adding a note into CHANGELOG.md?

Also, something to maybe be aware of: we'll change to Git.jl for `git` (#2348). I assume it should _just work_ on Woodpecker, but you never know.

**Comment by uncomfyhalomacro** [comment]
> Apologies for the delay here. I haven't really tested this, but the changes LGTM, and the test suite looks really comprehensive. Thank you!
> 
> It's ready to be merged, right? Would you mind adding a note into CHANGELOG.md?
> 
> Also, something to maybe be aware of: we'll change to Git.jl for `git` (#2348). I assume it should _just work_ on Woodpecker, but you never know.

It should probably just work assuming `git` and `Git.jl` *have* 1:1 compatibility. I will fix if it breaks Woodpecker 😄

**Comment by uncomfyhalomacro** [comment]
It seems I forgot that 2.0.0 was released recently. That was fast. I guess I will update the docs and the links which causes linkchecks to fail.

**Comment by uncomfyhalomacro** [comment]
\cc @mortenpi ready for merging. thank you!

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Documenter now warns when it encounters invalid keys in the various key-value at-blocks. ([#2306], [#2324])
 * File sizes are now expressed in more human-readable format. ([#2272], [#2344])
 * Documenter now uses [Git.jl](https://github.com/JuliaVersionControl/Git.jl) (as opposed to the system's `git` binary) for Git operations. ([#2348])
+* Woodpecker CI 1.x and 2.x are now also supported. ([#2339])
```
Comment (mortenpi) [inline CHANGELOG.md:15]: I updated this phrasing a bit -- we're also supporting 2.x, right?

**Comment by mortenpi** [comment]
Thanks for iterating on this @uncomfyhalomacro!

**Pull request merged**